src/Shared/Extensions/ClaimsPrincipalExtensions.cs

```
@@ -24,6 +25,7 @@ public static int GetUserId(this ClaimsPrincipal principal)
         }
 
         var claim = principal.FindFirst(x => x.Type.Equals(ClaimTypes.NameIdentifier));
+
         if (claim == null)
         {
             throw new InvalidOperationException("The 'sub' claim is not present in the principal.");
```

It's not necessary at the moment, but it's better to move exception messages to a separate file

---

What if I need to throw an exceptions with interpolated string messages or create a class only for parameterless messages?

---

You can create custom exceptions and overload their constructors with various parameters. Within the service, you simply call the required exception constructor, eliminating the need to explicitly write the message. For constant messages, you can create a separate class and directly insert them into the exception